lib/plugins/aws/deploy-function.js

```diff
@@ -267,11 +267,40 @@ class AwsDeployFunction {
     if (functionObj.layers && !functionObj.layers.some(_.isObject)) {
       params.Layers = functionObj.layers;
     }
+
+    // We need to initialize to an empty array so if a layer is removed
+    // we will send an empty Layers array in the update call to remove any layers.
+    // If there are no layers in the remove config this property will be set to undefined anyway.
+    if (!params.Layers) {
+      params.Layers = [];
+    }
+    if (!remoteFunctionConfiguration.Layers) {
+      remoteFunctionConfiguration.Layers = [];
+    }
+
+    // Check if we have remotely managed layers and add them to the update call
+    // if they exist in the remote function configuration.
+    const isConsoleSdkLayerArn = RegExp.prototype.test.bind(
+      /(?:177335420605|321667558080):layer:sls-/u
+    );
+    const serverlessConsoleLayerArns = (remoteFunctionConfiguration.Layers || [])
+      .filter(({ Arn: arn }) => isConsoleSdkLayerArn(arn))
+      .map(({ Arn }) => Arn);
+    const hasServerlessConsoleLayers = serverlessConsoleLayerArns.length > 0;
+    if (params.Layers && serverlessConsoleLayerArns.length > 0) {
```

Comment:
`param.Layers` is truthy here unconditionally, so it's dead check.

Also it's better to use `hasServerlessConsoleLayers` that we've defined above, instead of again checking `serverlessConsoleLayerArns.length > 0`